homotopy-web/src/app/debug_gl.rs

```diff
@@ -0,0 +1,262 @@
+use std::{cell::RefCell, rc::Rc};
+
+use homotopy_common::hash::FastHashSet;
+use homotopy_core::{debug::Drawable, Diagram};
+use homotopy_gl::{
+    array::VertexArray,
+    draw,
+    frame::{DepthTest, Frame},
+    vertex_array, GlCtx, Result,
+};
+use homotopy_graphics::geom::{CubicalGeometry, SimplicialGeometry};
+use ultraviolet::{Vec3, Vec4};
+use web_sys::CanvasRenderingContext2d;
+use yew::prelude::*;
+
+use super::diagram_gl::{renderer::shaders::Shaders, OrbitCamera};
+use crate::{
+    buffers::buffer_debug,
+    components::{
+        delta::{Delta, DeltaAgent},
+        touch_interface::TouchInterface,
+    },
+    two_d::TwoDCtx,
+};
+
+pub struct DebugScene {
+    // rendered by 2D canvas context
+    pub vertex_labels: Vec<(Vec3, String)>,
+    // rendered by GL
+    pub wireframe_components: Vec<VertexArray>,
+}
+
+impl DebugScene {
+    pub fn new(ctx: &GlCtx, diagram: &Diagram) -> Result<Self> {
+        let cubical = match diagram.dimension() {
```

Comment:
`CubicalGeometry::new` isn't really designed for this. What you want is a different constructor that makes a geometry containing all the points/lines but none of areas/volumes. That would be easy to write, and you wouldn't even need to use the `Mesh`, you can just use the exploded graph (and configure that to ignore the boundary).